test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +91,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (new Parser().parseCommand(inputCommand).isMutating()){
```

Comment:
why need a new instance? 
Does this way test if non-mutating commands skip save?